src/app/dataset-view/additional-components/dataset-settings-component/tabs/webhooks/create-edit-subscription-modal/create-edit-subscription-modal.component.ts

```diff
@@ -0,0 +1,112 @@
+/**
+ * Copyright Kamu Data, Inc. and contributors. All rights reserved.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the LICENSE file.
+ */
+
+import { ChangeDetectionStrategy, Component, inject, Input, OnInit } from "@angular/core";
+import { AbstractControl, FormBuilder, Validators } from "@angular/forms";
+import { NgbActiveModal } from "@ng-bootstrap/ng-bootstrap";
+import AppValues from "src/app/common/values/app.values";
+import {
+    DatasetBasicsFragment,
+    WebhookSubscriptionInput,
+    WebhookSubscriptionStatus,
+} from "src/app/api/kamu.graphql.interface";
+import { WebhooksService } from "src/app/services/webhooks.service";
+import { BaseComponent } from "src/app/common/components/base.component";
+import { takeUntilDestroyed } from "@angular/core/rxjs-interop";
+import {
+    CreateWebhookSubscriptionSuccess,
+    SubscribedEventType,
+    WebhookSubscriptionModalAction,
+} from "./create-edit-subscription-modal.model";
+import { MaybeNull, MaybeUndefined } from "src/app/interface/app.types";
+import { Clipboard } from "@angular/cdk/clipboard";
+import { changeCopyIcon } from "src/app/common/helpers/app.helpers";
+import { ErrorSets } from "src/app/common/directives/form-validation-errors.types";
+
+@Component({
+    selector: "app-create-edit-subscription-modal",
+    templateUrl: "./create-edit-subscription-modal.component.html",
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class CreateEditSubscriptionModalComponent extends BaseComponent implements OnInit {
+    @Input({ required: true }) public datasetBasics: DatasetBasicsFragment;
+    @Input() public subscriptionData: MaybeNull<CreateWebhookSubscriptionSuccess>;
+
+    public dropdownList: SubscribedEventType[] = [];
+    public secret: MaybeUndefined<string>;
+    public readonly WebhookSubscriptionStatus: typeof WebhookSubscriptionStatus = WebhookSubscriptionStatus;
+    public readonly WebhookSubscriptionModalAction: typeof WebhookSubscriptionModalAction =
+        WebhookSubscriptionModalAction;
+    public readonly ErrorSets: typeof ErrorSets = ErrorSets;
+
+    public activeModal = inject(NgbActiveModal);
+    private fb = inject(FormBuilder);
+    private webhooksService = inject(WebhooksService);
+    private clipboard = inject(Clipboard);
+
+    public ngOnInit(): void {
+        this.webhooksService
+            .eventTypes()
+            .pipe(takeUntilDestroyed(this.destroyRef))
```

Comment:
The list of event type never changes. You should query it only once, not on every modal.

Reply:
2 times the result will be returned from the GQL cache